test/utils/string_extension_test.dart

```diff
@@ -377,4 +378,90 @@ void main() {
       expect('<a href="https://example.com"></a>'.extractInnerText(), isEmpty);
     });
   });
+
+  group('buildHighlightTextSpans tests', () {
+    test(
+        'buildHighlightTextSpans handles special * and \\ characters in highlightText',
+        () {
+      const text =
+          'This is a test string with special characters like * and \\.';
+      const highlightText = '* and \\';
+      final expectedSpans = [
+        const TextSpan(
+          text: 'This is a test string with special characters like ',
+          style: TextStyle(color: Colors.black),
+        ),
+        const TextSpan(text: '* and \\', style: TextStyle(color: Colors.red)),
+        const TextSpan(text: '.', style: TextStyle(color: Colors.black)),
+      ];
+
+      final result = text.buildHighlightTextSpans(
+        highlightText,
+        style: const TextStyle(color: Colors.black),
+        highlightStyle: const TextStyle(color: Colors.red),
+      );
+
+      expect(result.length, expectedSpans.length);
+
+      for (int i = 0; i < result.length; i++) {
+        expect(result[i].text, expectedSpans[i].text);
+        expect(result[i].style, expectedSpans[i].style);
+      }
+    });
+
+    test(
+        'buildHighlightTextSpans handles special \\ characters in highlightText',
+        () {
+      const text = 'This is a test string with special characters like 123\\.';
+      const highlightText = '123';
+      final expectedSpans = [
+        const TextSpan(
+          text: 'This is a test string with special characters like ',
+          style: TextStyle(color: Colors.black),
+        ),
+        const TextSpan(text: '123', style: TextStyle(color: Colors.red)),
+        const TextSpan(text: '\\.', style: TextStyle(color: Colors.black)),
+      ];
+
+      final result = text.buildHighlightTextSpans(
+        highlightText,
+        style: const TextStyle(color: Colors.black),
+        highlightStyle: const TextStyle(color: Colors.red),
+      );
+
+      expect(result.length, expectedSpans.length);
+
+      for (int i = 0; i < result.length; i++) {
+        expect(result[i].text, expectedSpans[i].text);
+        expect(result[i].style, expectedSpans[i].style);
+      }
+    });
+
+    test('buildHighlightTextSpans handles special characters in highlightText',
+        () {
+      const text = 'This is a test string with special characters like 123.';
+      const highlightText = '123';
```

Comment:
how about if the highlightText is `123.` or `123\\`?

Reply:
I added more test cases for this case.

For demo UI:

![Simulator Screenshot - iPhone 15 - 2024-08-12 at 11 16 44](https://github.com/user-attachments/assets/8f46cc1d-0d15-44ad-93e5-480028145e77)

![Simulator Screenshot - iPhone 15 - 2024-08-12 at 11 16 23](https://github.com/user-attachments/assets/639457fe-af8c-4e11-be77-0de44ea7d092)